frontend_egui/src/app.rs

```diff
@@ -389,7 +393,23 @@ impl SnowGui {
                     ui.close_menu();
                 }
                 ui.separator();
+                ui.menu_button("Recent workspaces", |ui| {
+                    for (i, path, display_name) in self.settings.get_recent_workspaces_for_display() {
+                        if ui
+                            .button(format!("{}: {}", i, display_name))
+                            .clicked()
+                        {
+                            self.load_workspace(Some(&path));
+                            self.update_titlebar(ctx);
+                            ui.close_menu();
+                        }
+                    }
+                });
+                ui.separator();
                 if ui.button("Exit").clicked() {
+                    if let Err(e) = self.settings.save() {
```

Comment:
This will not be called if the user closes the window using the close/X button. Maybe just try to save when changes to the config occur (even in add_recent_workspace)?